File: src/1039-find-the-town-judge/solution.c
#include <stdio.h>

int findJudge(int n, int** trust, int trustSize, int* trustColSize) {

    int f[n+1];
    int t[n+1];

    for(int i = 1; i <= n; i++) {
        f[i] = t[i] = 0;
    }

    for(int i = 0; i < trustSize; i++) {
        f[trust[i][0]]++;
        t[trust[i][1]]++;
    }

    for(int i = 1; i <= n; i++) {
        if (f[i] == 0 && t[i] == n - 1)
            return i; 
    }

    return -1;
}

File: src/1039-find-the-town-judge/solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        trusted, trusts = [0] * n, [0] * n

        for [a, b] in trust:
            trusted[b - 1] += 1
            trusts[a - 1] += 1

        for i in range(n):
            if trusted[i] == n - 1 and trusts[i] == 0:
                return i + 1
        
        return -1
